src/commands/kv/bucket/upload.rs

```diff
@@ -23,5 +29,83 @@ pub fn upload(
         Err(e) => failure::bail!("{}", e),
     };
 
-    put_bulk(target, user, namespace_id, pairs?)
+    let mut pairs = pairs?;
+    validate_file_uploads(pairs.clone())?;
+
+    // Create a vector of uploads; that is, a vector of vectors of key-value pairs, each of which are
+    // maximum 10K key-value pairs in size OR maximum 100MB in size.
+    let mut key_count = 0;
+    let mut key_pair_bytes = 0;
+    let mut key_value_batch: Vec<KeyValuePair> = Vec::new();
+
+    while !(pairs.is_empty() && key_value_batch.is_empty()) {
+        if pairs.is_empty() {
+            // Last batch to upload
+            call_put_bulk_api(target, user.clone(), namespace_id, &mut key_value_batch)?;
+        } else {
+            let pair = pairs.pop().unwrap();
```

Comment:
Can't use `?` here because the error type from `pop()` doesn't work with this failure crate trait :(

Reply:
womp womp